### PR TITLE
Add Xavier and Kaiming Initializations

### PR DESCRIPTION
**Pull request by Chanlaw**
# Description
Update initializations for `HookedTransformer` to support xavier/kaiming uniform/normal. MuP weight inits are supported but not 1/d_head attention or Adam learning rate rescaling (will be added soon). Also I haven't touched `HookedTransformerEncoder`, which seems to require a larger refactor. 

Note that default init is still "gpt2", so we don't break existing functionality. @jbloomAus should we just update the init to Kaiming normal, which I think is the best of the non-MuP initializations? 

The important stuff are in the `utils.py`, `test_utils.py`, and `HookedTransformer.py` files, the rest is mainly lot of small unimportant so the various linters aren't mad.

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update

# Checklist:

- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] I have not rewritten tests relating to key interfaces which would affect backward compatibility

### CODE REVIEW

**Comment by jbloomAus** [comment]
Happy to update the default initialization strategy. I think ~95% of T-lens users are using pre-trained models anyway. As long as the docs are updated to reflect the default init, I think that's fine.

**Comment by jbloomAus** [comment]
Accepting now and can change default in a follow up.

**Comment by Chanlaw** [comment]
Sounds good, thanks for the quick response!

Yeah, most people are just using it to poke at pretrained 100m-10b models.

My main concern with changing the default is that it might screw over the various algorithmic task tutorials that involving training from scratch (esp the grokking ones, where weight norm at initialization sometimes matter). Happy to include the change when I'm done adding in MuP though.
